lib/src/package_config.dart

```diff
@@ -274,3 +276,24 @@ String extractLanguageVersion(VersionConstraint c) {
   }
   return '${minVersion.major}.${minVersion.minor}';
 }
+
+/// Parses the (now obsolete) .packages file.
+Map<String, Uri> parsePackagesFile(List<int> bytes, Uri base) {
+  final result = <String, Uri>{};
+  for (var line in utf8.decode(bytes).split('\n')) {
+    line = line.trim();
+    final commentLocation = line.indexOf('#');
+    if (commentLocation != -1) line = line.substring(0, commentLocation);
```

Comment:
I think `#` is only allowed on the first character of a line, see:
https://github.com/dart-lang/package_config/blob/master/lib/packages_file.dart#L33-L97

Interpreting it as comment elsewhere is probably a mistake.